tests/ducktape/test_producer_with_schema_registry.py

```diff
@@ -0,0 +1,286 @@
+            else:
+                # Calculate actual latency if we have send time
+                msg_key = msg.key().decode('utf-8', errors='replace') if msg.key() else 'unknown'
+                if msg_key in send_times:
+                    latency_ms = (time.time() - send_times[msg_key]) * 1000
+                    del send_times[msg_key]  # Clean up
+                else:
+                    latency_ms = 0.0  # Default latency if timing info not available
+
+                metrics.record_delivered(latency_ms, topic=msg.topic(), partition=msg.partition())
+
+        # Start metrics collection
+        metrics.start()
+
+        self.logger.info("Producing messages with %s serialization and metrics for %.1f seconds to topic %s",
+                         serialization_type, test_duration, topic_name)
+        start_time = time.time()
+        messages_sent = 0
+
+        while time.time() - start_time < test_duration:
+            message_value = message_value_func(messages_sent)
+            message_key = str(uuid4())
+
+            try:
+                # Calculate message size for metrics
+                serialized_key = string_serializer(message_key)
+                serialized_value = serializer(message_value, SerializationContext(topic_name, MessageField.VALUE))
+                message_size = len(serialized_key) + len(serialized_value)
+
+                # Record message being sent with metrics
+                metrics.record_sent(message_size, topic=topic_name, partition=0)
+
+                # Track send time for latency calculation
+                send_times[message_key] = time.time()
+
+                producer.produce(
+                    topic=topic_name,
+                    key=serialized_key,
+                    value=serialized_value,
+                    callback=delivery_callback
+                )
+                messages_sent += 1
+
+                # Poll frequently to trigger delivery callbacks and record poll operations
+                if messages_sent % 100 == 0:
+                    producer.poll(0)
+                    metrics.record_poll()
+
+            except BufferError:
+                # Record buffer full events and poll
+                metrics.record_buffer_full()
+                producer.poll(0.001)
+                continue
+
+        # Flush to ensure all messages are sent
+        self.logger.info("Flushing producer...")
+        producer.flush(timeout=30)
+
+        # Finalize metrics collection
+        metrics.finalize()
+
+        # Get comprehensive metrics summary and validate
+        metrics_summary = metrics.get_summary()
+        self.calculate_and_verify_results(metrics_summary, bounds, serialization_type)
+
+    def test_basic_produce_with_avro_serialization(self):
+        """Test producing messages with Avro serialization using Schema Registry"""
+        if Producer is None:
+            self.logger.error("confluent_kafka not available, skipping test")
+            return
```

Comment:
BTW pytest has decorators for skipping tests conditionally which is better than returning early if you need this pattern in the future